processing/src/main/java/io/druid/query/QueryRunnerFactory.java

```diff
@@ -57,7 +57,7 @@
    * @param queryRunners Individual QueryRunner objects that produce some results
    * @return a QueryRunner that, when asked, will use the ExecutorService to runt he base QueryRunners
    */
-  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners);
+  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners, QueryType query);
```

Comment:
Here we're introducing the concept of context for the merge task. The query runner factory hasn't had context itself prior to this. Context is often used within the runner itself. I don't have a good solution off the top of my head yet, but are we sure we want to introduce context at this level?